sonar-php-plugin/src/main/java/org/sonar/plugins/php/XmlProfileParser.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition.NewBuiltInActiveRule;
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition.NewBuiltInQualityProfile;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+final class XmlProfileParser {
+
+  private XmlProfileParser() {
+  }
+
+  static void parse(String path, NewBuiltInQualityProfile profile) {
+    try {
+      String content = toString(path);
+      XmlFile xmlFile = XmlFile.create(content);
+
+      NodeList ruleNodeList = xmlFile.getDocument().getElementsByTagName("rule");
+      XmlFile.asList(ruleNodeList).forEach(ruleNode -> {
+        Element ruleElement = (Element) ruleNode;
+        String repositoryKey = getChildContent(ruleElement, "repositoryKey");
+        String key = getChildContent(ruleElement, "key");
+
+        NewBuiltInActiveRule newActiveRule = profile.activateRule(repositoryKey, key);
+
+        NodeList parameterNodeList = ruleElement.getElementsByTagName("parameter");
+        XmlFile.asList(parameterNodeList).forEach(parameter -> {
+          Element parameterElement = (Element) parameter;
+          String paramKey = getChildContent(parameterElement, "key");
+          String paramValue = getChildContent(parameterElement, "value");
+          newActiveRule.overrideParam(paramKey, paramValue);
+        });
+
+        Optional<String> priority = getOptionalChildContent(ruleElement, "priority");
+        priority.ifPresent(newActiveRule::overrideSeverity);
+      });
+
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to create profile", e);
+    }
+  }
+
+  private static String getChildContent(Element element, String tagName) {
+    Optional<String> optionalChildContent = getOptionalChildContent(element, tagName);
```

Comment:
This could be replaced by a orElseThrow call : https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-

But I wonder about the added value of using optional instead of a simple null check in this context as you only wrap into an optional to unwrap it on the very next call.